### DISCUSSION TITLE
C#中使用Process调用FFmpeg相关问题

### DISCUSSION THREAD

**Discussion by AlvinRey** [BYJRK/DotNet-Discussions#192] [C# 技术交流]
先上代码，问题在下面。
![PixPin_2023-12-26_14-42-13](https://github.com/BYJRK/DotNet-Discussions/assets/56423507/a923e925-281f-4be1-8e78-53134889766a)
FFmpeg将处理过程的信息都输出到StandardError流上，我通过ErrorDataReceived异步去处理这些信息，到目前为止都没什么问题。
在命令行运行FFmpeg时，当已存在与输出文件同名的文件时，FFmpeg会提示是否覆盖，然后需要用户键入[y/N]在确认是否覆盖，如下图
![image](https://github.com/BYJRK/DotNet-Discussions/assets/56423507/196a0508-cb9c-4c5a-a3d1-504face30ff7)

问题就是这个询问信息为什么Process的StandardError流捕获不到？下图是IDE的Debug信息，对比上图可以看到，就只有询问那一句没有被输出。
![image](https://github.com/BYJRK/DotNet-Discussions/assets/56423507/4e65d974-0172-4098-b28a-d363a85a8970)

**Comment by AlvinRey** [comment]
OK，似乎在StackOverFlow上找到答案了。File ... already exists. Overwrite? [y/N]这个用户提示信息是直接修改屏幕内存，不是走StandardError流所以捕获不到。

**Reply by BYJRK** [reply]
那这怎么解决呢？

**Reply by AlvinRey** [reply]
只能在调用的时候顺带输入 -y或-n这两个argument之一，以保证不会出现File ... already exists. Overwrite? [y/N]使进程死锁（因为通过Process调用FFmpeg一般都会将CreateNoWindow设置为true，这样就不能手动选择，进程就会一直卡选择。）至于检查重名，要在调用FFmpeg前通过其他方法检测是否存在同名文件，然后如果存在则通过弹窗让用户选择是否覆盖。

**Reply by BYJRK** [reply]
原来如此，学到了
